### PR TITLE
Smoke, dust, and MYNN updates

### PR DESCRIPTION
**Pull request by SamuelTrahanNOAA**
RRFS-Smoke is improved, cleaned, and rebranded as RRFS-SD. Now with smoke, dust, and coarsepm tracer handling, it adds dry deposition velocity to MYNN-EDMF, updates the smoke/dust direct and indirect feedback. There are bug fixes:  switch to non-aerosol-aware in the GF scheme and correct emission bug in FENGSHA.

Also included are MYNN updates from #43

These changes are from @haiqinli and @joeolson42.

Related issue: https://github.com/ufs-community/ufs-weather-model/issues/1659

### CODE REVIEW

File: physics/rrtmgp_aerosol_optics.meta
Comment (dustinswales) [inline physics/rrtmgp_aerosol_optics.meta]: I propose we change the standard_name to "aerosol_optical_depth_at_550nm"

File: physics/sgscloud_radpre.F90
Comment (dustinswales) [inline physics/sgscloud_radpre.F90:37]: This module can use some housekeeping wrt the working precision used throughout. As a follow up PR.

File: physics/smoke_dust/module_smoke_plumerise.F90
Comment (dustinswales) [inline physics/smoke_dust/module_smoke_plumerise.F90]: Please remove if unused. Github history will document these changes.

File: physics/smoke_dust/module_smoke_plumerise.F90
Comment (dustinswales) [inline physics/smoke_dust/module_smoke_plumerise.F90]: ditto

File: physics/smoke_dust/rrfs_smoke_wrapper.F90
Comment (dustinswales) [inline physics/smoke_dust/rrfs_smoke_wrapper.F90:24]: To avoid ambiguities it's best to explicitly state what to import by using ONLY: ...

File: physics/smoke_dust/rrfs_smoke_postpbl.F90
Comment (dustinswales) [inline physics/smoke_dust/rrfs_smoke_postpbl.F90:14]: To avoid ambiguities it's best to explicitly state what to import by using ONLY: ...

**Review by dustinswales — CHANGES_REQUESTED** [review]
Looks good.
Some minor suggestions.

File: physics/rrtmgp_aerosol_optics.meta
Comment (haiqinli) [reply physics/rrtmgp_aerosol_optics.meta]: Hi Dustin. There's already 2D AOD variable in UFS, which has been implemented for physics to account for impact of the climatological aerosols.
Actually here we add 3D aerosol extinction field, which will be used for the surface visibility, and other diagnostics. We could name this variable as "aerosol_extinction_at_550nm". Any thoughts?
Thanks.

File: physics/sgscloud_radpre.F90
Comment (haiqinli) [reply physics/sgscloud_radpre.F90:37]: We will with Joe on this one.

File: physics/smoke_dust/module_smoke_plumerise.F90
Comment (haiqinli) [reply physics/smoke_dust/module_smoke_plumerise.F90]: Sure, we have removed these unused lines.

File: physics/smoke_dust/module_smoke_plumerise.F90
Comment (haiqinli) [reply physics/smoke_dust/module_smoke_plumerise.F90]: Sure, we have also removed these unused lines.

File: physics/rrtmgp_aerosol_optics.meta
Comment (dustinswales) [reply physics/rrtmgp_aerosol_optics.meta]: [The existing 2D AOD is vertically integrated, whereas this AOD being introduced is by layer](https://github.com/SamuelTrahanNOAA/ccpp-physics/blob/ufs/dev-rrfs-sd/physics/radiation_aerosols.f#L4490). Both are using the [same SW band](https://github.com/SamuelTrahanNOAA/ccpp-physics/blob/ufs/dev-rrfs-sd/physics/radiation_aerosols.f#L443), so I think the 2D AOD diagnostic can be derived from this new diagnostic?

Taking a step back, I believe the field you are adding as a diagnostic to the setaer() routine is already being output as aerosw? 
aerosw contains the by-band aerosol optical depth at each layer, for which you [just need a single band](https://github.com/SamuelTrahanNOAA/ccpp-physics/blob/ufs/dev-rrfs-sd/physics/radiation_aerosols.f#L443). aerosw has the standard_name aerosol_optical_depth_for_shortwave_bands_01_16, so  aerosol_optical_depth_at_550nm is named similar.

Comment (yangfanglin) [reply physics/rrtmgp_aerosol_optics.meta]: adding @AnningCheng-NOAA  and @kate.zhang@noaa.gov for their comments.  I recall there was a correction made to write out AOD at 550nm.

Comment (haiqinli) [reply physics/rrtmgp_aerosol_optics.meta]: Hi Fanglin and Dustin, thanks for your comments. We need the 3D aerosol extinction at 550nm for surface visibility diagnostics, and this is based on the correction from Anning and Kate with tauae_550. As Anning and Kate pointed out that aerosw is the average between band intervals, and they introduced tauae_550 just at 550nm. tauae_550 at each layer is assigned to ext550. We will update the PR with the name of "aerosol_optical_depth_at_550nm" as Dustin suggested.

4488 ! --- update diagnostic aod arrays
4489           do k = 1, NLAY
4490             aerodp(i,1) = aerodp(i,1) + tauae_550(k,1)
4491             ext550(i,k) = tauae_550(k,1)

File: physics/smoke_dust/rrfs_smoke_wrapper.F90
Comment (haiqinli) [reply physics/smoke_dust/rrfs_smoke_wrapper.F90:24]: We will update the PR to explicitly import by using ONLY. Thanks.

File: physics/smoke_dust/rrfs_smoke_postpbl.F90
Comment (haiqinli) [reply physics/smoke_dust/rrfs_smoke_postpbl.F90:14]: We will update the PR to explicitly import by using ONLY. Thanks.

File: physics/rrtmgp_aerosol_optics.meta
Comment (SamuelTrahanNOAA) [reply physics/rrtmgp_aerosol_optics.meta]: I have merged haiqinli's branch that changes this to `aerosol_optical_depth_at_550nm`

**Comment by SamuelTrahanNOAA** [comment]
@dustinswales - I just merged a branch from @haiqinli that should resolve the problems you found, other than the one you said should be fixed in a follow-up PR instead.

File: physics/GFS_rrtmg_pre.F90
Comment (grantfirl) [inline physics/GFS_rrtmg_pre.F90]: Are any of these hard-coded constants tunable? If so, consider making them variables.

Comment (SamuelTrahanNOAA) [reply physics/GFS_rrtmg_pre.F90]: This is specifically designed to divide by 3 and do a unit conversion, so putting them in variables doesn't make much sense.

Comment (haiqinli) [reply physics/GFS_rrtmg_pre.F90]: This is tunable. The value was based on our 10-days retro runs. I will try to update it to variables.

Comment (grantfirl) [reply physics/GFS_rrtmg_pre.F90]: It's not a huge deal, I'm just averse to seeing hard-coded data be changed in-code in subsequent PRs, especially if folks reading the code have no idea what the numbers represent.

File: physics/module_bl_mynn.F90
Comment (grantfirl) [inline physics/module_bl_mynn.F90:1]: @SamuelTrahanNOAA @haiqinli Please tell me that all of these changes are already in https://github.com/ufs-community/ccpp-physics/pull/43 so that I can skip them.

Comment (haiqinli) [reply physics/module_bl_mynn.F90:1]: Yes, all the MYNN changes were merged from PR #43, and you can skip them. Thanks.

File: physics/module_mp_nssl_2mom.F90
Comment (grantfirl) [inline physics/module_mp_nssl_2mom.F90]: Ha. Agreed. Why can't a debug flag be used to control this output? Also, probably don't need to keep the comment in the repo.

File: physics/module_bl_mynn.F90
Comment (SamuelTrahanNOAA) [reply physics/module_bl_mynn.F90:1]: "git diff" agrees with Haiqin: the branches have identical module_bl_mynn.F90 files.

`git diff joe/mynn_01March..HEAD -- physics/module_bl_mynn.F90`

File: physics/mynnedmf_wrapper.F90
Comment (grantfirl) [inline physics/mynnedmf_wrapper.F90:1]: Same comment with this file -- I assume that these changes come from #43, although there are 423 lines changed here vs 424 changed in #43. I wonder what's going on there.

File: physics/module_mp_nssl_2mom.F90
Comment (SamuelTrahanNOAA) [reply physics/module_mp_nssl_2mom.F90]: Yes, I can use `ndebug`. I'm compiling that now to make sure it works.

Comment (haiqinli) [reply physics/module_mp_nssl_2mom.F90]: We temporally added these when we debugged the MYNN coupling to NSSL MP in PR #43. I will restore it in this PR.

File: physics/mynnedmf_wrapper.F90
Comment (SamuelTrahanNOAA) [reply physics/mynnedmf_wrapper.F90:1]: There are intentional changes in this file. From `git diff joe/mynn_01March..HEAD -- physics/mynnedmf_wrapper.F90`:

```diff
diff --git a/physics/mynnedmf_wrapper.F90 b/physics/mynnedmf_wrapper.F90
index 1e8eabe9..83a73e6b 100644
--- a/physics/mynnedmf_wrapper.F90
+++ b/physics/mynnedmf_wrapper.F90
@@ -156,7 +156,7 @@ SUBROUTINE mynnedmf_wrapper_run(        &
      &  imp_physics_thompson, imp_physics_wsm6,            &
      &  imp_physics_fa,                                    &
      &  chem3d, frp, mix_chem, rrfs_sd, enh_mix,           &
-     &  nchem, ndvel,                                      &
+     &  nchem, ndvel, vdep, smoke_dbg,                     &
      &  imp_physics_nssl, nssl_ccn_on,                     &
      &  ltaerosol, mraerosol, spp_wts_pbl, spp_pbl,        &
      &  lprnt, huge, errmsg, errflg                        )
@@ -181,6 +181,7 @@ SUBROUTINE mynnedmf_wrapper_run(        &
      !smoke/chem
      integer, intent(in) :: nchem, ndvel
      integer, parameter  :: kdvel=1
+     logical, intent(in) :: smoke_dbg
 
 ! NAMELIST OPTIONS (INPUT):
      logical, intent(in) ::                                 &
@@ -287,10 +288,9 @@ SUBROUTINE mynnedmf_wrapper_run(        &
 !smoke/chem arrays
       real(kind_phys), dimension(:), intent(inout) :: frp
       logical, intent(in) :: mix_chem, enh_mix, rrfs_sd
-      logical, parameter  :: smoke_dbg = .false. !set temporarily
       real(kind_phys), dimension(:,:,:), intent(inout) :: chem3d
+      real(kind_phys), dimension(:,:  ), intent(inout) :: vdep
       real(kind_phys), dimension(im)   :: emis_ant_no
-      real(kind_phys), dimension(im,ndvel) :: vdep
 
 !MYNN-2D
       real(kind_phys), dimension(:), intent(in) ::                       &
@@ -357,7 +357,6 @@ SUBROUTINE mynnedmf_wrapper_run(        &
 
       !initialize arrays for test
       EMIS_ANT_NO = 0.
-      vdep        = 0.
 
       FLAG_OZONE = ntoz>0
```

File: physics/module_mp_nssl_2mom.F90
Comment (SamuelTrahanNOAA) [reply physics/module_mp_nssl_2mom.F90]: @haiqinli - don't restore the line. I'm adding `ndebug>1` to it, and I'll merge that change in a few minutes.

File: physics/mynnedmf_wrapper.F90
Comment (SamuelTrahanNOAA) [reply physics/mynnedmf_wrapper.F90:1]: The `vdep`-related changes are necessary to prevent a crash when FV3_HRRR is used without rrfs_sd. This wasn't a problem before since you had to use the FV3_HRRR_smoke suite to get smoke, and there isn't much of a point in disabling smoke if you're explicitly trying to use its suite.

Comment (grantfirl) [reply physics/mynnedmf_wrapper.F90:1]: OK, @SamuelTrahanNOAA thanks for doing a diff for me.

File: physics/module_mp_nssl_2mom.F90
Comment (SamuelTrahanNOAA) [reply physics/module_mp_nssl_2mom.F90]: The gnu all suites build passed, so I pushed this change.

File: physics/smoke_dust/rrfs_smoke_wrapper.F90
Comment (grantfirl) [inline physics/smoke_dust/rrfs_smoke_wrapper.F90]: Redefining (and hard-coding) physical constants is against CCPP rules. I suggest to use `gas_constant_of_dry_air` and `ratio_of_vapor_to_dry_air_gas_constants_minus_one` from the host's physical constants.

Comment (grantfirl) [reply physics/smoke_dust/rrfs_smoke_wrapper.F90]: You're already passing in g, pi, ...

**Comment by grantfirl** [comment]
Nice work @haiqinli and @joeolson42 -- only minor comments really. There is room for computational improvement with dividing by physical constants within N-nested loops, but I'm not going to hold up your progress for that (and it might be optimized by the compiler anyway?). I'll approve once comments are addressed.

File: physics/smoke_dust/rrfs_smoke_wrapper.F90
Comment (haiqinli) [reply physics/smoke_dust/rrfs_smoke_wrapper.F90]: I will update to use gas_constant_of_dry_air and ratio_of_vapor_to_dry_air_gas_constants_minus_one from the host's physical constants. Thanks for your suggestions.

**Comment by SamuelTrahanNOAA** [comment]
@grantfirl - @haiqinli is preparing some changes to address your comments and I'll merge them from his branch as soon as possible.

File: physics/smoke_dust/rrfs_smoke_postpbl.F90
Comment (grantfirl) [inline physics/smoke_dust/rrfs_smoke_postpbl.F90:1]: @haiqinli @SamuelTrahanNOAA Can you remind me why this separate interstitial is necessary, rather than using GFS_PBL_generic_post? Is it because suites with MYNN EDMF don't use it?

Comment (haiqinli) [reply physics/smoke_dust/rrfs_smoke_postpbl.F90:1]: @grantfirl Because GFS_PBL_generic_post is not used in the FV3_HRRR physics suite. Thus we add it separately for the RRFS-SD with FV3_HRRR physis suite.

**Comment by SamuelTrahanNOAA** [comment]
@grantfirl - I just merged changes from @haiqinli to fix the problems you found, and I also updated to the top of ufs/dev.

**Review by grantfirl — APPROVED** [review]
Thanks for the prompt updates @haiqinli @SamuelTrahanNOAA .

**Comment by grantfirl** [comment]
@dustinswales You'll need to re-review since you requested changes.

**Review by dustinswales — APPROVED** [review]
Looks good. Thanks for making the requested changes.

**Comment by jkbk2004** [comment]
@grantfirl test was one on https://github.com/ufs-community/ufs-weather-model/pull/1658. we can merge in this pr.

**Review by Qingfu-Liu — APPROVED** [review]

**Pull request merged**
